### ISSUE TITLE
Add custom types

### ISSUE THREAD

**Issue by ahoarau** [cbandera/rosparam_handler#61]
That would really awesome if we could have something like this : 
```python

# gen.addcustomType(type,class_name,lambda);

gen = ParameterGenerator()
gen.addCustomType("std::vector<double>","Eigen::VectorXd",Array,"[](std::vector<double> array){ return Eigen::Map<Eigen::VectorXd>(array.data(),array.size());  }")
```

That would generate something like : 
```c++
Eigen::VectorXd __param_name;
std::function<Eigen::VectorXd(std::vector<double>)> array_to__param_name = [](std::vector<double> array){ return Eigen::Map<Eigen::VectorXd>(array.data(),array.size());  };

void fromYaml()
{
    // load the array as usual
  __param_name = array_to__param_name(   array_from_yaml);
}



```

**Comment by ahoarau** [comment]
The `std::vector<double>` could even be hidden in `gen.addArrayType` to be able to transform to geometry_msgs, tf msgs, eigen, etc.

**Comment by artivis** [comment]
Hi, this is an interesting idea !
Adding support for Eigen is on the todo list since a while (something similar to [rosparam_shortcuts](https://github.com/PickNikRobotics/rosparam_shortcuts)) however I couldn't get started due to a lack of time.
Your proposal (use of a lambdas) is interesting in that it generalizes well, however I'm concerned by the implications in terms of dependencies (missing includes etc) that it adds.
Maybe you could propose some proof-of-concept implementation so that we can discuss it further and see how rosparam_handler could be made even more generic !
Cheers.

**Comment by ahoarau** [comment]
Includes could be added as an extra argument (easy), but you should make sure to **link** against the said library. Making sure this happens is much more difficult. Maybe out of scope as well.

**Comment by artivis** [comment]
You are right, although Eigen being template only, this particular dependency could be handled similarly to what's already done with dynamic_reconfigure (see e.g. [here](https://github.com/cbandera/rosparam_handler/blob/develop/templates/Parameters.h.template#L14) or [there](https://github.com/cbandera/rosparam_handler/blob/develop/templates/Parameters.h.template#L57)).
